src/seccomp/src/lib.rs

```diff
@@ -1116,64 +1212,58 @@ fn EXAMINE_SYSCALL() -> Vec<sock_filter> {
     )]
 }
 
-/// Possible errors that could be encountered while processing a seccomp level value or generating
-/// a BPF program based on it.
+/// Possible errors that could be encountered while generating a BPF program.
 #[derive(Debug)]
 pub enum SeccompError {
     /// Error while trying to generate a BPF program.
     SeccompFilter(Error),
-    /// Failed to parse to `u8`.
-    Parse(std::num::ParseIntError),
-    /// Seccomp level is an `u8` value, other than 0, 1 or 2.
-    Level(u8),
 }
 
 impl Display for SeccompError {
     fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
         match *self {
             SeccompError::SeccompFilter(ref err) => write!(f, "Seccomp error: {}", err),
-            SeccompError::Parse(ref err) => write!(f, "Could not parse to 'u8': {}", err),
-            SeccompError::Level(arg) => write!(
-                f,
-                "'{}' isn't a valid value for 'seccomp-level'. Must be 0, 1 or 2.",
-                arg
-            ),
         }
     }
 }
 
-/// Possible values for seccomp level.
-#[repr(u8)]
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub enum SeccompLevel {
-    /// Seccomp filtering disabled.
-    None = 0,
-    /// Level of filtering that causes only syscall numbers to be examined.
-    Basic = 1,
-    /// Level of filtering that causes syscall numbers and parameters to be examined.
-    Advanced = 2,
+/// Binary Deserialization errors.
+#[derive(Debug)]
+pub enum DeserializationError {
+    /// Error when doing bincode deserialization.
+    Bincode(BincodeError),
 }
 
-impl SeccompLevel {
-    /// Converts from a seccomp level value of type String to the corresponding SeccompLevel variant
-    /// or returns an error if the parsing failed.
-    pub fn from_string(seccomp_value: String) -> std::result::Result<Self, SeccompError> {
-        match seccomp_value.parse::<u8>() {
-            Ok(0) => Ok(SeccompLevel::None),
-            Ok(1) => Ok(SeccompLevel::Basic),
-            Ok(2) => Ok(SeccompLevel::Advanced),
-            Ok(level) => Err(SeccompError::Level(level)),
-            Err(err) => Err(SeccompError::Parse(err)),
+impl Display for DeserializationError {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        use self::DeserializationError::*;
+
+        match *self {
+            Bincode(ref err) => write!(f, "Bincode deserialization failed: {}", err),
         }
     }
 }
 
+/// Deserialize a BPF file into a collection of usable BPF filters.
+pub fn deserialize_binary(
+    reader: &mut dyn Read,
+) -> std::result::Result<BpfThreadMap, DeserializationError> {
+    Ok(
+        bincode::deserialize_from::<&mut dyn Read, BpfThreadMap>(reader)
```

Comment:
I'm still concerned that this allows unlimited reading into memory, if presented with a huge file to read from.